src/include/miopen/logger.hpp

```diff
@@ -342,6 +343,33 @@ std::string LoggingParseFunction(const char* func, const char* pretty_func);
         std::cerr << miopen_driver_cmd_ss.str();                                        \
     } while(false)
 
+class LogScopeTime
+{
+    public:
+    LogScopeTime(std::string name)
+        : m_name(std::move(name)), m_beg(std::chrono::high_resolution_clock::now())
+    {
+    }
+    ~LogScopeTime()
+    {
+        auto end = std::chrono::high_resolution_clock::now();
+        auto dur = std::chrono::duration_cast<std::chrono::microseconds>(end - m_beg);
+        std::ostringstream ss;
+        ss << "MIOpenLogScopeTime: " << m_name << " : " << dur.count() << " us";
+        MIOPEN_LOG_I2(ss.str());
+    }
+
+    private:
+    std::string m_name;
+    std::chrono::time_point<std::chrono::high_resolution_clock> m_beg;
+};
+
+#if MIOPEN_LOG_FUNC_TIME_ENABLE
```

Comment:
[Recommendation] move this `#if` just above line 346 to hide `class LogScopeTime` when it is unnecessary.

Reply:
updated